app/serializers/rest/application_serializer.rb

```diff
@@ -19,6 +19,18 @@ def client_secret
     object.secret
   end
 
+  def redirect_uri
+    # Doorkeeper stores the redirect_uri value as a newline delimeted list in
+    # the database, as we may have more than one redirect URI, return the first:
+    object.redirect_uri.split.first
+  end
```

Comment:
This method / property should probably be considered deprecated, because previously it *could* return a value that included a new line character, which wouldn't be a valid URI, if the application was created not through the API.

i.e., if you go to the Development > Applications section in the settings, then the redirect_uris is a textarea allowing newline characters; It just so happens that we've no way to request that application back through the API to hit the previously faulty code here.

Reply:
@ClearlyClaire up to you as to what we do with the behaviour here: do we do a "correct" behaviour, or do we retain the previous behaviour?

Reply:
I *think* I would keep the existing behavior, however weird it is.

Reply:
Have retained that behaviour in the latest commit.